File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/domain/oid4vci/IssuerMetadata.kt
package my.ssdid.sdk.domain.oid4vci

import kotlinx.serialization.json.JsonObject

data class IssuerMetadata(
    val credentialIssuer: String,
    val credentialEndpoint: String,
    val credentialConfigurationsSupported: Map<String, JsonObject>,
    val tokenEndpoint: String,
    val authorizationEndpoint: String?
)

File: sdk/ios/SsdidCore/Sources/SsdidCore/Domain/OpenId4Vci/IssuerMetadata.swift
import Foundation

/// Issuer metadata resolved from well-known endpoints.
public struct IssuerMetadata: @unchecked Sendable {
    public let credentialIssuer: String
    public let credentialEndpoint: String
    public let credentialConfigurationsSupported: [String: [String: Any]]
    public let tokenEndpoint: String
    public let authorizationEndpoint: String?

    public init(credentialIssuer: String, credentialEndpoint: String, credentialConfigurationsSupported: [String: [String: Any]], tokenEndpoint: String, authorizationEndpoint: String?) {
        self.credentialIssuer = credentialIssuer
        self.credentialEndpoint = credentialEndpoint
        self.credentialConfigurationsSupported = credentialConfigurationsSupported
        self.tokenEndpoint = tokenEndpoint
        self.authorizationEndpoint = authorizationEndpoint
    }
}
